src/commands/dev/edge/server/https.rs

```diff
@@ -0,0 +1,105 @@
+use crate::commands::dev::utils::get_path_as_str;
+use crate::terminal::{emoji, message};
+
+use std::sync::{Arc, Mutex};
+
+use chrono::prelude::*;
+use futures_util::stream::StreamExt;
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Client as HyperClient, Request, Server};
+use hyper_rustls::HttpsConnector;
+use tokio::net::TcpListener;
+
+pub async fn https(
+    server_config: ServerConfig,
+    preview_token: Arc<Mutex<String>>,
+    host: String,
+) -> Result<(), failure::Error> {
+    tls::generate_cert()?;
+
+    // set up https client to connect to the preview service
+    let https = HttpsConnector::new();
+    let client = HyperClient::builder().build::<_, Body>(https);
+
+    // create a closure that hyper will use later to handle HTTP requests
+    let service = make_service_fn(move |_| {
+        let client = client.to_owned();
+        let preview_token = preview_token.to_owned();
+        let host = host.to_owned();
+
+        async move {
+            Ok::<_, failure::Error>(service_fn(move |req| {
+                let client = client.to_owned();
+                let preview_token = preview_token.lock().unwrap().to_owned();
+                let host = host.to_owned();
+                let version = req.version();
+                let (parts, body) = req.into_parts();
+                let req_method = parts.method.to_string();
+                let now: DateTime<Local> = Local::now();
+                let path = get_path_as_str(&parts.uri);
+                async move {
+                    let resp = preview_request(
+                        Request::from_parts(parts, body),
+                        client,
+                        preview_token.to_owned(),
+                        host.clone(),
+                        false,
+                    )
+                    .await?;
+
+                    println!(
+                        "[{}] {} {}{} {:?} {}",
+                        now.format("%Y-%m-%d %H:%M:%S"),
+                        req_method,
+                        host,
+                        path,
+                        version,
+                        resp.status()
+                    );
+                    Ok::<_, failure::Error>(resp)
+                }
+            }))
+        }
+    });
+
+    let listening_address = server_config.listening_address;
+
+    let mut tcp = TcpListener::bind(&listening_address).await?;
+    let tls_acceptor = &tls::get_tls_acceptor()?;
+    let incoming_tls_stream = tcp
+        .incoming()
+        .filter_map(move |s| async move {
+            let client = match s {
+                Ok(x) => x,
+                Err(e) => {
+                    println!("Failed to accept client");
+                    return Some(Err(e));
+                }
+            };
+            match tls_acceptor.accept(client).await {
+                Ok(x) => Some(Ok(x)),
+                Err(e) => {
+                    println!("Client connection error {}", e);
```

Comment:
```suggestion
                    eprintln!("Client connection error {}", e);
```

Reply:
when i go to `https://127.0.0.1:8787` in my browser i get `ERR_SSL_KEY_USAGE_INCOMPATIBLE` instead of being able to click through a warning and i get this in my `dev` session:

```
[2020-08-10T17:39:41Z ERROR rustls::session] TLS alert received: Message {
        typ: Alert,
        version: TLSv1_3,
        payload: Alert(
            AlertMessagePayload {
                level: Fatal,
                description: IllegalParameter,
            },
        ),
    }
Client connection error received fatal alert: IllegalParameter
```

Reply:
I didn't catch this earlier because firefox didn't check `KeyUsage` in the cert while chrome did. Should be fixed now